src/core/index.js

```diff
@@ -86,7 +86,18 @@ class IPFS extends EventEmitter {
     this._libp2pNode = undefined
     this._bitswap = undefined
     this._blockService = new BlockService(this._repo)
-    this._ipld = new Ipld(this._blockService)
+
+    // Make sure IPLD has the correct BlockService
+    const ipldDefaults = {
+      blockService: this._blockService
+    }
+    if (this._options.ipld === undefined) {
+      this._options.ipld = ipldDefaults
+    } else if (this._options.ipld.blockService === undefined) {
+      this._options.ipld.blockService = ipldDefaults.blockService
+    }
+    this._ipld = new Ipld(this._options.ipld)
```

Comment:
I've put your code in a file:

```js
'use strict'

const ipldDagPb = 'ipldDagPb'
const _options = {ipld: {blockService: 'passed in block service'}}
const _blockService = 'blockService'

const result = Object.assign(
  {},
  { formats: [ipldDagPb] },
  _options.ipld,
  { blockService: _blockService }
)

console.log(result)
```

The out put is `{ formats: [ 'ipldDagPb' ], blockService: 'blockService' }`, but I would expect it to be 
`{ formats: [ 'ipldDagPb' ], blockService: 'passed in block service' }`

Reply:
this is correct for js-ipfs! we need to force the already instantiated blockservice or we will need to deal with the repo instantiation. 

ipld blockservice option is an instance and not a reference, and blockservice needs a repo, but ipfs already creates a repo in the instantiation/init/start lifecycle. This will get complex very fast lol :)

i think we should force blockservice and maybe in a future PR handle all the blockservice/repo instances sync with ipld config